airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2827,13 +2827,9 @@ def create_record_selector(
             else None
         )
 
-        if model.transform_before_filtering is None:
-            # default to False if not set
-            model.transform_before_filtering = False
-
-        assert model.transform_before_filtering is not None  # for mypy
```

Comment:
This whole structure of an assertion in the existing code is very weird. nice to get rid of an unnecessary assert statement